packages/gatsby-plugin-manifest/src/__tests__/gatsby-node.js

```diff
@@ -209,8 +208,31 @@ describe(`Test plugin manifest options`, () => {
       }
     )
 
-    expect(sharp).toHaveBeenCalledTimes(2)
+    expect(sharp).toHaveBeenCalledTimes(3)
     const content = JSON.parse(fs.writeFileSync.mock.calls[0][1])
     expect(content).toEqual(manifestOptions)
   })
+
+  it(`icon options iterator adds options and the icon array take precedence`, async () => {
+    fs.statSync.mockReturnValueOnce({ isFile: () => true })
+
+    const pluginSpecificOptions = {
+      icon: `images/gatsby-logo.png`,
+      icon_options: {
+        purpose: `maskable`,
+      },
+    }
+    await onPostBootstrap(
+      { reporter },
+      {
+        ...manifestOptions,
+        ...pluginSpecificOptions,
+      }
+    )
+
+    expect(sharp).toHaveBeenCalledTimes(3)
+    const content = JSON.parse(fs.writeFileSync.mock.calls[0][1])
```

Comment:
Not new here (so this is a nit pick!), but couldn't this be written as:

```js
expect(fs.writeFileSync).toHaveBeenCalledWith(expect.any(String), [
  expect.objectContaining({
    purpose: 'all'
  }),
  expect.objectContaining({
    purpose: 'maskable'
  })
], expect.any(String))
```

Something to consider for the future, but does not need to be implemented here!

Reply:
Probably, I will not claim to be an 'expect' expert.